_studio/mfx_lib/scheduler/src/mfx_scheduler_core_ischeduler.cpp

```diff
@@ -362,63 +361,50 @@ mfxStatus mfxSchedulerCore::WaitForTaskCompletion(const void *pOwner)
         WakeUpThreads();
     }
 
+    std::list<mfxTaskHandle> tasks;
 
-    do
     {
-        // reset waiting handle
-        waitHandle.handle = 0;
-
-        // enter protected section.
-        // make searching in a separate code block,
-        // to avoid dead-locks with other threads.
-        {
-            std::lock_guard<std::mutex> guard(m_guard);
-            int priority;
+        mfxTaskHandle waitHandle;
+        std::lock_guard<std::mutex> guard(m_guard);
 
-            priority = MFX_PRIORITY_HIGH;
-            do
+        ForEachTask(
+            [&](MFX_SCHEDULER_TASK* task)
```

Comment:
looks like too big closure. I believe you need only ```tasks&```

Reply:
and pOwner

Reply:
ok. will slightly change